woocommerce/woocommerce
# fix - Product Button: don't trigger jQuery add to cart function when the button is clicked on classic themes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





### Why

The issue happens because the script `add-to-cart` is enqueued. For this reason, when the button is clicked:
- the interactivity API code is triggered.
- the `add-to-cart` script is triggered.

For this reason, the product is added twice. While on the block themes, we can avoid enqueueing the `add-to-cart` script with the `wp_enqueue_scripts` action; this does not work on classic themes. I suspect the page is generated differently, so it is too late when the logic inside the block is executed.

For this reason:
- I added the class `wc-interactivity` on the button
- I updated the selector on the `add-to-cart` script to avoid running it on the buttons that have the wc-interactivity class.

Also, I added some E2E tests to increase the coverage.


Closes #43288.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Switch to Storefront.
2. Create a new post and add the Product Collection block.
3. Save and visit the post.
4. Click on the add to cart button and ensure the product is added once.
5. Visit the `/shop` page and add a product.
6. Ensure that the product is added correctly.
7. Switch to a block theme (TT4, for example).
8. Visit the post created earlier.
9. Click the add to cart button to ensure the product is added once.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
Fix - Product Button: don't trigger old add to cart function when the button is clicked on classic themes

#### Comment 

</details>
